agent/client/client.go

```diff
@@ -138,3 +140,7 @@ func (c *Client) getName() (string, error) {
 
 	return hostname, nil
 }
+
+func isCancelledError(err error) bool {
```

Comment:
I didn't find a good way of doing this, the rpc message contains a wrapped error that when I tried to unwrap it it was returning a plane `interface{}` object 🤷🏾

Reply:
Did you try: 

```
errors.Is(err, context.Canceled)
```